rasa/core/policies/unexpected_intent_policy.py

```diff
@@ -4,6 +4,11 @@
 from pathlib import Path
 from typing import Any, List, Optional, Text, Dict, Type, Union, TYPE_CHECKING
 
+try:
+    from typing import TypedDict
+except ImportError:
+    from mypy_extensions import TypedDict
```

Comment:
why not doing it as in the other modules?

```
if TYPE_CHECKING:
    from typing_extensions import TypedDict
```

Reply:
`mypy_extensions` is currently not even in our dependencies

Reply:
But don't you want to always import it somehow from one of the packages. I am not sure on when `TYPE_CHECKING` is set to `True` and when is it set to `False`, hence the question.

Reply:
`TYPE_CHECKING` is only true when we use `mypy` to analyze the code and during development in your IDE. As `typing_extension` is currently a dev package you need the `if` part (plus quotes around the type when using it). Good point that we should re-consider moving that our production packages 🤔

Reply:
> TYPE_CHECKING is only true when we use mypy to analyze the code and during development in your IDE

If that's the case, then you can't import it inside an `if` statement. You always want it to be imported somehow since the code needs to initialize a `TypedDict` and it will result in a runtime error if the `TypedDict` is never imported.

Reply:
Yep, the tests start [failing](https://github.com/RasaHQ/rasa/pull/9016/checks?check_run_id=2987874145) because of that then.

Reply:
Should we do - 

```
try:
    from typing import TypedDict
except ImportError:
    from typing_extensions import TypedDict
```
to keep support for python 3.6 and 3.7

Reply:
Hmm, I used quotes but type checks are still [failing](https://github.com/RasaHQ/rasa/pull/9016/checks?check_run_id=2988025301#step:11:16) 😞

Reply:
> If that's the case, then you can't import it inside an if statement. You always want it to be imported somehow since the code needs to initialize a TypedDict and it will result in a runtime error if the TypedDict is never imported.

You also need to put the `TypedDict` stuff in the `if`. See [here](https://github.com/RasaHQ/rasa/blob/8db9030bf7ec457e9306c95cbf7fbe811771d05f/rasa/core/test.py#L66)